software/contrib/menu.md

```diff
@@ -0,0 +1,141 @@
+
+## Menu Inclusion
+
+In order to be included in the menu a program needs to meet a few additional requirements, described below. Programs are
+not required to participate in the menu, but if they do not, they should be added to the ``EXCLUDED_SCRIPTS`` list near 
+the top of [``menu.py``](/software/contrib/menu.py).
+
+An example 'hello world' script is shown here and the details discussed below.
+
+``hello_world.py``
+```Python
+import uasyncio as asyncio  # 1
+from europi import oled
+
+
+async def main():  # 2
+    while True:
+        oled.centre_text("Hello world")
+        await asyncio.sleep(1)  # 3
+
+
+asyncio.run(main())  # 4
+
+```
+
+1. **Import ``asyncio``:** MicroPython includes an asyncio library that is similar to the library included in CPython 
+under the name ``uasyncio``. These libraries are similar enough that it is standard practice to import ``uasyncio`` 
+under the alias ``asyncio``.
+
+2. **An asynchronous main method:** The main code of the script should be encapsulated in an asynchronous method so that
+you can await on any sleep calls, described below. The method itself does not need to be called 'main'.
+
+3. **``sleep()`` inside any loops:** In order for the user to be able to request that your program exit, your program should 
+periodically relinquish control to other threads. This is most easily done by adding calls to `asyncio.sleep()` to any
+loops that will run for a significant amount of time. The program's main loop is the most obvious place for this; but,
+depending on your code, there may be other loops that also execute for a significant amount of time which should also
+sleep.
+
+4. **Execute your main method asynchronously:** Use ``asyncio.run()`` to call your main method asynchronously. This
+allows the menu program to properly call your script and wait for its completion.
+
+## Support testing
+
+To support unit testing in CPython, a couple of changes need to be made to the example ``hello_world.py``. In order to have something to test, we'll add the display of a simple counter to our script, a method to increment it, and add a test
+that the increment method works properly.
+
+``hello_world.py``
+```python
+try:  # 1
+    import uasyncio as asyncio
+except ImportError:
+    import asyncio
+from europi import oled
+
+
+def increment(counter):
+    return counter + 1
+
+
+async def main():
+    counter = 0
+    while True:
+        oled.centre_text(f"Hello world\n{counter}")
+
+        counter = increment(counter)
+
+        await asyncio.sleep(1)
+
+
+if __name__ in ["__main__", "contrib.hello_world"]:  # 2
+    asyncio.run(main())
+
+```
+
+``test_hello_world.py``
+```python
+from hello_world import increment
+
+
+def test_increment():
+    assert increment(1) == 2
+```
+
+1. **Import ``uasyncio`` but fall back to ``asyncio``:** Since ``uasyncio`` doesn't exist in CPython, it will raise an 
```

Comment:
I _think_ this may be unnecessary. I noticed in the MicroPython 1.17 release notes:

> u-module references have been replaced with just the module name without the u-prefix to help clear up the intended usage of modules in MicroPython.

https://github.com/micropython/micropython/releases/tag/v1.17

Reply:
Hmm, nope. Just tested this and got:
```
Traceback (most recent call last):
  File "main.py", line 5, in <module>
ImportError: no module named 'asyncio'
```